src/Workspaces/Core/Portable/Shared/Extensions/SyntaxGeneratorExtensions_CreateEqualsMethod.cs

```diff
@@ -350,10 +346,14 @@ private static bool IsPrimitiveValueType(ITypeSymbol typeSymbol)
                     case SpecialType.System_Single:
                     case SpecialType.System_Double:
                     case SpecialType.System_String:
-                    case SpecialType.System_Nullable_T:
                     case SpecialType.System_DateTime:
                         return true;
                 }
+
+                if (typeSymbol.IsNullable(out var underlyingType))
+                {
+                    return ShouldUseEqualityOperator(underlyingType);
+                }
```

Comment:
💡 You can place this before the preceding switch statement.

```csharp
if (typeSymbol.IsNullable(out var underlyingType))
{
  // Some explanation goes here...
  typeSymbol = underlyingType;
}
```

Reply:
nit: could simplify to: `return typeSymbol.IsNullable(...) && ShouldUse...`

Reply:
@CyrusNajmabadi My nit would be the other way: don't write the last check in a different form than the other checks because it's not fundamentally different :D